src/discord/censor.py

```diff
@@ -123,8 +129,23 @@ async def __censor(self, message: discord.Message):
 
         # Make sure pinging through @everyone, @here, or any role can not happen
         mention_perms = discord.AllowedMentions(everyone=False, users=True, roles=False)
+        if reply:
+
+            _message = (
+                reply.cached_message
+                if reply.cached_message
+                else await self.bot.get_channel(reply.channel_id).fetch_message(
+                    reply.message_id
+                )
+            )
+            reply_message = (
+                f"*Replying to {_message.author.mention}*"
+                f"\n>[{reply.cached_message.content[:40]}...]({reply.cached_message.jump_url})\n"
+            )
+        else:
+            reply_message = ""
         await webhook.send(
-            content,
+            (content + reply_message)[:2000] if reply_message else content,
```

Comment:
We shouldn't be truncating messages to 2000 chars. A Nitro user can post up to a 4000 char message which means this would chop half of their message.

If we are doing this method, we should be splitting the compiled message up into separate messages and send those, but even then another person's messages could be spliced in between.